Blog for I Just Want Authentication To Work (#2598)
FusionAuth/fusionauth-site: astro/src/content/blog/i-just-want-authentication-to-work.mdx

---
publish_date: 2023-10-19
title: I Just Want Authentication To Work
description: When I am writing an Application the last thing that I care about is setting up Authentication, I just want it to work!
authors: Alex Patterson
image: 
categories: Company, News
tags: award, security mag
excerpt_separator: "{/* more */}"
---

Authentication is the process of verifying the identity of a user or entity. It is a fundamental security measure that is used in a wide variety of applications, including web applications, mobile apps, and APIs. The thing is I just don't care about figuring it out! I just want it to work!!

{/* more */}

> There are many different ways to implement authentication, but the most common approach is to use a username and password. Yep, just like back in 1960 when [The Everly Brothers](https://youtu.be/KtY3L5tDN_E?si=BATB3BzJJFpWsHmB) had one of the top songs out there, this method is still most popular today. In my opinion, just like I have never heard this song before, I don't want to hear about passwords any more either!

Reviewer: I basically want to say that the song is old and not relevant just like passwords should be (or will be)

Another common authentication method is to use a single sign-on (SSO) solution. SSO solutions allow users to log in to multiple applications using a single set of credentials. This can be convenient for users, as they do not have to remember multiple usernames and passwords. This method came about around the same time Eminem hit the [billboard top 100](https://youtu.be/6mPCy9LMxTI?si=5A2I-N1neb_8rGOl) so at least this is in my wheelhouse.

> My point is that Authentication is an important security measure, but it can also be complex to implement. This is why we have been hanging on to these old methods so long! When you are creating a new application (unless it is one for Authentication) the last thing you care about is trying to roll your own. There are many different authentication methods to choose from, and each method has its own advantages and disadvantages. Additionally, authentication needs to be implemented in a secure way to prevent attackers from gaining unauthorized access to applications. When all I care about creating is the next [Napster](https://youtu.be/CKrdsGdLVQ8?si=bTp4DHKS4tlpQzfW) so everyone can share their 1990's music with me again!

Reviewer 2: "...the last thing you care about is trying to roll your own. "  I think this should be more along the lines of 
"the last thing you want to do is spend precious development cycles trying to roll your own security when you can get it for [free](https://fusionauth.io/download)."  or something like that.

* Reduce the risk of compliance violations: Authentication services can help you to meet compliance requirements by providing features such as two-factor authentication and user provisioning.

The last thing that I want is a call on Saturday night that someone hacked our password database that we built in house and didn't secure!

![Meme you can't leak passwords if you don't store passwords, man pointing at head](/img/blogs/i-just-want-authentication-to-work/meme-leak.png)

## Choosing an authentication service

When choosing an authentication service, there are a few factors to consider:

* Features: Consider the features that are important to you, such as two-factor authentication, user provisioning, and support for different authentication methods.
* Pricing: Authentication services can vary in price. Compare the pricing of different services to find the best deal for your needs.
* Ease of use: Consider how easy the authentication service is to use. You want a service that is easy to set up and manage.
* Hosting: Do you need flexibility to host your own instance of an Authentication Provider or have them Host.

> ## For developers who don't care about implementing authentication

Reviewer: I still think the whole stype thing is so dumb it is literally `capitalize` in tailwind.

> If you are a developer who doesn't care about implementing authentication, there are a few things that you can do to make your life easier:

Reviewer 2: Again, they probably care, but don't want to do it themselves.

* Use an authentication service. There are many different authentication services available, both free and paid. Authentication services can save you the hassle of having to implement and maintain your own authentication system.

> * Use a framework or library that provides authentication functionality. There are many different frameworks and libraries available that provide authentication functionality. Using a framework or library can save you the time and effort of having to implement authentication yourself. If you are ready to get started checkout our [quickstarts](/docs/quickstarts/).

Reviewer 2: may want to say FusionAuth Quickstarts.

I know you don't want it to be "salesy", but you haven't mentioned fusion auth at all to this point so "our quickstars" might not be clear.

> * Consider when that Napster, wait I mean your app starts to grow and now you need authentication for mobile, desktop and web. You will want an Authentication provider that can handle each form of Authentication.

Reviewer 2: ```suggestion
* Consider that when Better Napster, wait I mean your app starts to grow, you will want an Authentication provider that can handle different forms of Authentication such as mobile, desktop and web.
```

Reviewer 2: One of the things that I think is super useful about fusion auth is that it is super easy to implement other IdPs.  Google, Facebook, Apple, etc.

Another great thing about FusionAuth is that it can support multiple apps.  So not just Better Napster, but the Even Better Than That Napster.

I also think it is worth mentioning that FusionAuth can be deployed localy for development, hosted yourself or as a service and the code should work wherever you point it to.  Basically can do it all.

---
publish_date: 2023-10-19
title: I Just Want Authentication To Work
description: When I am writing an Application the last thing that I care about is setting up Authentication, I just want it to work!
authors: Alex Patterson
image: 
categories: Company, News
tags: award, security mag
excerpt_separator: "{/* more */}"
---

Authentication is the process of verifying the identity of a user or entity. It is a fundamental security measure that is used in a wide variety of applications, including web applications, mobile apps, and APIs. The thing is I just don't care about figuring it out! I just want it to work!!

{/* more */}

There are many different ways to implement authentication, but the most common approach is to use a username and password. Yep, just like back in 1960 when [The Everly Brothers](https://youtu.be/KtY3L5tDN_E?si=BATB3BzJJFpWsHmB) had one of the top songs out there, this method is still most popular today. Just as the Everly Brothers' time at the top of the charts has passed, so has the time for passwords. I don't want to hear about them any more!

> A more modern approach is to use a single sign-on (SSO) solution. SSO solutions allow users to log in to multiple applications using a single set of credentials. This can be convenient for users, as they do not have to remember multiple usernames and passwords. This method came about around the same time Eminem hit the [billboard top 100](https://youtu.be/6mPCy9LMxTI?si=5A2I-N1neb_8rGOl) for the first time. SSO often has the problem of including the first login requirement of using a username and password still to this day. Even Eminem made some changes and remains relevant!

Reviewer 2: SSO often has the problem of including the first l ... This transition could be smoother.  Is this still a problem? Is it fixed?  Maybe something like this.  SSO is not perfect.  For example it has the problem of including the first log.....to this day.  However, just as Eminem has made some changes to stay relevant, so is SSO.

## For Developers Who Just Want It To Work

If you are a developer who just wants authentication to work without implementing it yourself, there are a few things that you can do to make your life easier:

* Use an authentication service. There are many different authentication services available, both free and paid. Authentication services can save you the hassle of having to implement and maintain your own authentication system.
* Use a framework or library that provides authentication functionality. There are many different frameworks and libraries available that provide authentication functionality. Using a framework or library can save you the time and effort of having to implement authentication yourself. If you are ready to get started checkout our [FusionAuth Quickstarts](/docs/quickstarts/).
* Consider that when Better Napster, wait I mean your app starts to grow, you will want an Authentication provider that can handle different forms of Authentication such as mobile, desktop and web.
* Develop on a local instance or self contained envioronment that you can host yourself without risking data corruption in production.

> ## What Should You Care About If Not Authentication

Reviewer 2: Is this section new?  It doesn't really feel like this section belongs in this blog to me.  That could just be an opinion though.  I do think working the Auth as service video was good though.

Reviewer: I wanted to get some backlinks pulling through. For me this is what the post is about. Once you have Auth figured out there is so much more to try and figure out for web now. SHould I tie it in better somehow saying that?

Reviewer 3: Maybe

```suggestion
## What You _Should_ Care About
```

- Hosting - [Vercel](https://vercel.com/), [Netlify](https://www.netlify.com/), [Fly.io](https://fly.io/), [Cloudflare](https://www.cloudflare.com/)
- CMS: [Contentful](https://www.contentful.com/), [Quintype](https://www.quintype.com/), [Hygraph](https://hygraph.com/), [Ghost](https://ghost.org/publishers/), [Strapi](https://strapi.io/), [Sanity](https://www.sanity.io/), [Kontent](https://kontent.ai/), [Optimizely](https://www.optimizely.com/products/orchestrate/content-management/)

There are full backend solutions out there like [Firebase](https://firebase.google.com/), [Supabase](https://supabase.com/), and [Appwrite](https://www.appwrite.io). However, if you don't like some part of this stack then it becomes very difficult for you to decouple your application(s). In this video our Founder and CEO talks about [Authentication as a Microservice](https://youtu.be/SLc3cTlypwM?si=qFzJaeA9NTeXGSfz&t=834) and how to use a JWT in your APIs.

## Finally

> Okay now that I got that out of my system I am off to listen to [Connor Price](https://youtu.be/UekfKh59KOw?si=xTM0powkicUHG2Yo) and login to Amazon using my [face](https://www.aboutamazon.com/news/retail/amazon-passwordless-sign-in-passkey), because it is 2023 people!

Reviewer 2: may want to add a few links to our docs and stuff.

---
publish_date: 2023-10-19
title: I Just Want Authentication To Work
description: When I am writing an Application the last thing that I care about is setting up Authentication, I just want it to work!
authors: Alex Patterson
image: /img/blogs/i-just-want-authentication-to-work/i-just-want-auth-to-work.png
categories: Company, News
tags: award, security mag
excerpt_separator: "{/* more */}"
---

> Authentication is the process of verifying the identity of a user or entity. It is a fundamental security measure that is used in a wide variety of applications, including web applications, mobile apps, and APIs. The thing is I just don't care about figuring it out! I just want it to work!!

Reviewer 3: Who is "I" here? Knowing you are an experienced dev lets me know if I should value your opinion or not.

{/* more */}

There are many different ways to implement authentication, but the most common approach is to use a username and password. Yep, just like back in 1960 when [The Everly Brothers](https://youtu.be/KtY3L5tDN_E?si=BATB3BzJJFpWsHmB) had one of the top songs out there, this method is still most popular today. Just as the Everly Brothers' time at the top of the charts has passed, so has the time for passwords. I don't want to hear about them any more!

A more modern approach is to use a single sign-on (SSO) solution. SSO solutions allow users to log in to multiple applications using a single set of credentials. This can be convenient for users, as they do not have to remember multiple usernames and passwords. This method came about around the same time Eminem hit the [billboard top 100](https://youtu.be/6mPCy9LMxTI?si=5A2I-N1neb_8rGOl) for the first time. SSO often has the problem of including the first login requirement of using a username and password still to this day. Even Eminem made some changes and remains relevant!

> My point is that Authentication is an important security measure, but it can also be complex to implement. This is why we have been hanging on to these old methods so long! When you are creating a new application (unless it is one for Authentication) the last thing you care about is trying to roll your own. There are many different authentication methods to choose from, and each method has its own advantages and disadvantages. Additionally, authentication needs to be implemented in a secure way to prevent attackers from gaining unauthorized access to applications. All I care about is that my data is secure when creating the next [Napster](https://youtu.be/CKrdsGdLVQ8?si=bTp4DHKS4tlpQzfW) so everyone can share their 1990's music with me again!

Reviewer 3: Why is Authentication capitalized sometimes and not others? I'd standardize on lower case.

Reviewer: corrected those without titles or start of sentences

![Napster app from Wikipedia](/img/blogs/i-just-want-authentication-to-work/napster.png)

## The benefits of using an authentication service

There are many benefits to using an authentication service. Authentication services can help you to:

* Save time and money: Implementing and maintaining authentication can be time-consuming and expensive. Authentication services can save you the hassle of having to implement and maintain your own authentication system.
* Improve security: Authentication services can help you to improve the security of your applications by using best practices and the latest security technologies keeping up to date with the latest security patches.
* Reduce the risk of compliance violations: Authentication services can help you to meet compliance requirements by providing features such as two-factor authentication and user provisioning.

> The last thing that I want is a call on Saturday night that someone hacked our password database that we built in house and didn't secure!

Reviewer 3: ```suggestion
The last thing that I want is a call on Saturday night that someone hacked our password database that we built in house two years ago and didn't maintain securely!
```

This feels like a realer situation to me. If it doesn't address your point, no worries.

![Meme you can't leak passwords if you don't store passwords, man pointing at head](/img/blogs/i-just-want-authentication-to-work/meme-leak.png)

## Choosing an authentication service

When choosing an authentication service, there are a few factors to consider:

* Features: Consider the features that are important to you, such as two-factor authentication, user provisioning, and support for different authentication methods.
* Pricing: Authentication services can vary in price. Compare the pricing of different services to find the best deal for your needs.
* Ease of use: Consider how easy the authentication service is to use. You want a service that is easy to set up and manage.

> * Hosting: Do you need flexibility to host your own instance of an Authentication Provider or have them Host.

Reviewer 3: Why is `Host` capitalized?

> 

Reviewer 3: ```suggestion
* Integration support: Often times you'll want to use an authentication service as a central user data store. You'll want to make sure you can integrate custom applications, off the shelf commercial software and open source solutions with it.

```

## For Developers Who Just Want It To Work

If you are a developer who just wants authentication to work without implementing it yourself, there are a few things that you can do to make your life easier:

* Use an authentication service. There are many different authentication services available, both free and paid. Authentication services can save you the hassle of having to implement and maintain your own authentication system.

> * Use a framework or library that provides authentication functionality. There are many different frameworks and libraries available that provide authentication functionality. Using a framework or library can save you the time and effort of having to implement authentication yourself. If you are ready to get started checkout our [FusionAuth Quickstarts](/docs/quickstarts/).

Reviewer 3: You could also suggest people look to whatever framework they are using and use an auth framework built for that.

- Database - [PlanetScale](https://planetscale.com/), [AlloyDB for PostgreSQL](https://cloud.google.com/alloydb), [AWS Aurora Serverless](https://aws.amazon.com/rds/aurora/serverless/), [CockroachDB Serverless](https://www.cockroachlabs.com/blog/announcing-cockroachdb-serverless/), [EdgeDB](https://www.edgedb.com/), [Neon](https://neon.tech/), [Supabase](https://supabase.com/), [Timescale](https://www.timescale.com/), [Upstash](https://upstash.com/), [Grafbase](https://grafbase.com/), [Turso](https://turso.tech/), [Fauna](https://fauna.com/), [Ably](https://ably.com), [Convex](https://www.convex.dev/), [Hasura](https://hasura.io/), [liveblocks](https://liveblocks.io/)
- Hosting - [Vercel](https://vercel.com/), [Netlify](https://www.netlify.com/), [Fly.io](https://fly.io/), [Cloudflare](https://www.cloudflare.com/)
- CMS: [Contentful](https://www.contentful.com/), [Quintype](https://www.quintype.com/), [Hygraph](https://hygraph.com/), [Ghost](https://ghost.org/publishers/), [Strapi](https://strapi.io/), [Sanity](https://www.sanity.io/), [Kontent](https://kontent.ai/), [Optimizely](https://www.optimizely.com/products/orchestrate/content-management/)

> There are full backend solutions out there like [Firebase](https://firebase.google.com/), [Supabase](https://supabase.com/), and [Appwrite](https://www.appwrite.io). However, if you don't like some part of this stack then it becomes very difficult for you to decouple your application(s). In this video our Founder and CEO talks about [Authentication as a Microservice](https://youtu.be/SLc3cTlypwM?si=qFzJaeA9NTeXGSfz&t=834) and how to use a JWT in your APIs.

Reviewer 3: ```suggestion
There are full backend solutions out there like [Firebase](https://firebase.google.com/), [Supabase](https://supabase.com/), and [Appwrite](https://www.appwrite.io). However, if you don't like some part of these stacks then it becomes very difficult for you to decouple your application(s). In this video our Founder and CEO talks about [Authentication as a Microservice](https://youtu.be/SLc3cTlypwM?si=qFzJaeA9NTeXGSfz&t=834) and how to use a JWT in your APIs.
```

Suggest reading this post out aloud; that will catch this kind of grammar issue.

## Finally

> Okay now that I got that out of my system I am off to listen to [Connor Price](https://youtu.be/UekfKh59KOw?si=xTM0powkicUHG2Yo) and login to Amazon using my [face](https://www.aboutamazon.com/news/retail/amazon-passwordless-sign-in-passkey), because it is 2023 people!

Reviewer 3: ```suggestion
Okay now that I got that out of my system I am off to listen to [Connor Price](https://youtu.be/UekfKh59KOw?si=xTM0powkicUHG2Yo) and log in to Amazon using my [face](https://www.aboutamazon.com/news/retail/amazon-passwordless-sign-in-passkey), because it is 2023 people!
```

Verb is `log in`. Noun is `login`.

---
publish_date: 2023-11-17
title: I Just Want Authentication To Work

> description: When I am writing an Application the last thing that I care about is setting up Authentication, I just want it to work!

Reviewer 3: I haven't seen capital A application or authentication in the blog. Typically, when we use that in doc, we mean the Application configuration/object in FusionAuth. Is that intentional?

authors: Alex Patterson
image: /img/blogs/i-just-want-authentication-to-work/i-just-want-auth-to-work.png

> categories: Company, News

Reviewer 3: These probably are not correct.

Reviewer: how about community

Reviewer 3: You can look at the allowed categories here: https://github.com/FusionAuth/fusionauth-site/blob/master/.github/known-blog-categories.txt

Whatever you think fits works for me, I just was pretty sure this wasn't 'news' :)

> tags: award, security mag

Reviewer 3: These too.

excerpt_separator: "{/* more */}"
---

> Authentication is the process of verifying the identity of a user or entity. It is a fundamental security measure that is used in a wide variety of applications, including web applications, mobile apps, and APIs. The thing is as a developer with over 20 years of experience, I just don't care about figuring it out! I just want it to work!!

Reviewer 3: ```suggestion
Authentication is the process of verifying the identity of a user or entity. It is a fundamental security measure that is used in a wide variety of applications, including web applications, mobile apps, and APIs. The thing is, as a developer with over 20 years of experience, I just don't care about figuring it out! I just want it to work!!
```

{/* more */}

There are many different ways to implement authentication, but the most common approach is to use a username and password. Yep, just like back in 1960 when [The Everly Brothers](https://youtu.be/KtY3L5tDN_E?si=BATB3BzJJFpWsHmB) had one of the top songs out there, this method is still most popular today. Just as the Everly Brothers' time at the top of the charts has passed, so has the time for passwords. I don't want to hear about them any more!

A more modern approach is to use a single sign-on (SSO) solution. SSO solutions allow users to log in to multiple applications using a single set of credentials. This can be convenient for users, as they do not have to remember multiple usernames and passwords. This method came about around the same time Eminem hit the [billboard top 100](https://youtu.be/6mPCy9LMxTI?si=5A2I-N1neb_8rGOl) for the first time. SSO often has the problem of including the first login requirement of using a username and password still to this day. Even Eminem made some changes and remains relevant!

> My point is that authentication is an important security measure, but it can also be complex to implement. This is why we have been hanging on to these old methods so long! When you are creating a new application (unless it is one for authentication) the last thing you care about is trying to roll your own. There are many different authentication methods to choose from, and each method has its own advantages and disadvantages. Additionally, authentication needs to be implemented in a secure way to prevent attackers from gaining unauthorized access to applications. All I care about is that my data is secure when creating the next [Napster](https://youtu.be/CKrdsGdLVQ8?si=bTp4DHKS4tlpQzfW) so everyone can share their 1990's music with me again!

Reviewer 3: ```suggestion
My point is that authentication is an important security measure, but it can also be complex to implement. This is why we have been hanging on to these old methods so long! When you are creating a new application (unless it is an *authentication* application), the last thing you care about is user login. There are many different authentication methods to choose from, and each method has its own advantages and disadvantages. Additionally, authentication needs to be implemented in a secure way to prevent attackers from gaining unauthorized access to applications. 

But, again, all I care about is that my data is secure when creating the next [Napster](https://youtu.be/CKrdsGdLVQ8?si=bTp4DHKS4tlpQzfW) so everyone can share their 1990's music with me again!
```

![Napster app from Wikipedia](/img/blogs/i-just-want-authentication-to-work/napster.png)

## The benefits of using an authentication service

> There are many benefits to using an authentication service. authentication services can help you to:

Reviewer 3: ```suggestion
There are many benefits to using an authentication service. Authentication services can help you:
```

* Hosting: Do you need flexibility to host your own instance of an authentication Provider or have them host.
* Integration support: Often times you'll want to use an authentication service as a central user data store. You'll want to make sure you can integrate custom applications, off the shelf commercial software and open source solutions with it.  

## For Developers Who Just Want It To Work

If you are a developer who just wants authentication to work without implementing it yourself, there are a few things that you can do to make your life easier:

* Use an authentication service. There are many different authentication services available, both free and paid. Authentication services can save you the hassle of having to implement and maintain your own authentication system.
* Use a framework or library that provides authentication functionality. There are many different frameworks and libraries available that provide authentication functionality. Using a framework or library can save you the time and effort of having to implement authentication yourself. If you are ready to get started checkout our [FusionAuth Quickstarts](/docs/quickstarts/).

> * Consider that when Better Napster, wait I mean your app starts to grow, you will want an authentication provider that can handle different forms of authentication such as mobile, desktop and web.

Reviewer 3: ```suggestion
* Consider that when Better Napster, wait I mean your app, starts to grow, you will want an authentication provider that can handle different devices and form factors such as mobile, desktop and web clients.
```

> * Develop on a local instance or self contained envioronment that you can host yourself without risking data corruption in production.

Reviewer 3: ```suggestion
* Develop on a local instance or self contained environment that you can host yourself to make development easier.
```

> * Look to your favoriate framework provider and see what they are using and use an authentication framework built for that.

Reviewer 3: spelling. Can you please run this through a spellchecker?

Author: apparently my grammarly extensions wasn't running sorry about that!

## What You _Should_ Care About

Now that you have authentication figured out you can start to focus on the real requirements of building an application...

> Deciding the rest of your software:

Reviewer 3: ```suggestion
Picking the rest of your software components:
```

Reviewer 3: And where does writing the actual code fit in?

- CMS: [Contentful](https://www.contentful.com/), [Quintype](https://www.quintype.com/), [Hygraph](https://hygraph.com/), [Ghost](https://ghost.org/publishers/), [Strapi](https://strapi.io/), [Sanity](https://www.sanity.io/), [Kontent](https://kontent.ai/), [Optimizely](https://www.optimizely.com/products/orchestrate/content-management/)

There are full backend solutions out there like [Firebase](https://firebase.google.com/), [Supabase](https://supabase.com/), and [Appwrite](https://www.appwrite.io). However, if you don't like some part of these stacks then it becomes very difficult for you to decouple your application(s). In this video our Founder and CEO talks about [Authentication as a Microservice](https://youtu.be/SLc3cTlypwM?si=qFzJaeA9NTeXGSfz&t=834) and how to use a JWT in your APIs.  

## Finally

Okay now that I got that out of my system I am off to listen to [Connor Price](https://youtu.be/UekfKh59KOw?si=xTM0powkicUHG2Yo) and log in to Amazon using my [face](https://www.aboutamazon.com/news/retail/amazon-passwordless-sign-in-passkey), because it is 2023 people!  

If you want authentication to just work start using [FusionAuth today for free](/pricing)!

> Want to get jump started using your framework of choice checkout our [Quickstarts](/docs/quickstarts/)

Reviewer 3: ```suggestion
Want to get jump started using your framework of choice? Check out our [Quickstarts](/docs/quickstarts/).
```

---
publish_date: 2023-11-17
title: I Just Want Authentication To Work
description: When I am writing an application the last thing that I care about is setting up authentication, I just want it to work!
authors: Alex Patterson
image: /img/blogs/i-just-want-authentication-to-work/i-just-want-auth-to-work.png
categories: Community

> tags: community story

Reviewer 3: We typically have community stories tagged when a member of the community is interviewed. See https://fusionauth.io/blog/treefort-uses-fusionauth-for-all-auth for example.

Here are tags that I'd use:

authentication, application development, napster, music, 1990s.

Author: @mooreds should I do "Off Topic" for category?

Reviewer 3: Maybe "Education"? We don't have Off Topic for a category.

Here's the list it is checked against: https://github.com/FusionAuth/fusionauth-site/blob/master/.github/known-blog-categories.txt

## Finally

Okay, now that I got that out of my system I am off to listen to [Connor Price](https://youtu.be/UekfKh59KOw?si=xTM0powkicUHG2Yo) and log in to Amazon using my [face](https://www.aboutamazon.com/news/retail/amazon-passwordless-sign-in-passkey) because it is 2023 people!

> Oh yeah and now the actual coding begins!!

Reviewer 3: Love it
